core/src/test/java/org/elasticsearch/common/settings/SettingTests.java

```diff
@@ -154,6 +154,20 @@ public void testUpdaterIsIsolated() {
         assertNull(ab2.get());
     }
 
+    public void testDeprecatedSetting() {
+        final Setting<Boolean> deprecatedSetting = Setting.boolSetting("deprecated.foo.bar", false, Property.Deprecated);
+        final Settings settings = Settings.builder().put("deprecated.foo.bar", true).build();
+        final int iterations = randomIntBetween(1, 128);
```

Comment:
is the intention here not to be strict on testing more than one `get`? If so, should this be bumped to `randomIntBetween(2, 128)`?